docs/tutorial.md

```diff
@@ -146,35 +159,97 @@ More on [`bot.craft()`](http://mineflayer.prismarine.js.org/#/api?id=botcraftrec
-`bot.heldItem` property to get the name of the item held by the bot. `bot.heldItem` is an instance of [`Item`](https://github.com/PrismarineJS/prismarine-item/blob/master/README.md) or `null` if no item held.
+[`health`](http://mineflayer.prismarine.js.org/#/api?id=health) - event that listens for changes in health or food points.
+[`bot.consume()`](http://mineflayer.prismarine.js.org/#/api?id=botconsumecallback) - method that makes bot eat currently held item.
+[`bot.food`](http://mineflayer.prismarine.js.org/#/api?id=botfood) - property to check food points.
+`bot.heldItem` property that returns [`Item`](https://github.com/PrismarineJS/prismarine-item/blob/master/README.md) - held by the bot or `null`.
 
 ```js
-const minFoodPoints = 18;
+const minFoodPoints = 18
 
-bot.on("health", function () {
+bot.on('health', function () {
   // listen on any changes in health or food points.
+  const heldItemName = bot.heldItem.name
   if (bot.food < minFoodPoints) {
     // if the food points lower than 18, then
-    const heldItemName = bot.heldItem.name;
-
-    function onFinishEating(err) {
-      if (err) return bot.chat(err.message);
-      bot.chat(`I'm finish eating ${heldItemName}`);
-    }
-
-    bot.chat(`Eating ${heldItemName}`);
-    bot.consume(onFinishEating);
+    bot.chat(`Eating ${heldItemName}`)
+    bot.consume(onFinishEating)
     // `bot.consume()` method takes `onFinishEating` as callback, and execute `onFinishEating` when its finish or stumble an error.
   }
-});
+  function onFinishEating (err) {
+    if (err !== null) {
+      bot.chat(err.message)
+      return
+    }
+    bot.chat(`I'm finish eating ${heldItemName}`)
+  }
+})
 ```
 
-#### Fisherman Bot
+## Advanced
+
+For the following, we assume you have understood the [Basics](#basics) tutorial.
+
+### Creating an event from chat
+
+You can create your own event from chat using [`bot.chatAddPattern()`](http://mineflayer.prismarine.js.org/#/api?id=botchataddpatternpattern-chattype-description) method. Useful for bukkit servers where the chat format changes a lot.
+[`bot.chatAddPattern()`](http://mineflayer.prismarine.js.org/#/api?id=botchataddpatternpattern-chattype-description) method takes three argumens :
+
+- `pattern` - regular expression (regex) to match chat
+- `chatType` - the event the bot emits when the pattern matches. Eg: "chat" or "whisper"
+- `description` - Optional, describes what the pattern is for
+
+You can add [Groups and Range](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Groups_and_Ranges) into the `pattern`, then the listener will spread the captured groups into arguments of your callback sequentially.
+
+Read more about [regular expression](https://en.wikipedia.org/wiki/Regular_expression).
+
+Examples :
+
+#### Answer Hello Bot
+
+Here we're creating a bot that answer 'hello' from the other player.
+
+```js
+bot.chatAddPattern(
+  /(helo|hello|Hello)/,
```

Comment:
`/hell?o/i` pretty much matches the same

Reply:
Okay